qualtran/bloqs/basic_gates/x_basis.py

```diff
@@ -84,7 +84,13 @@ def as_cirq_op(
         import cirq
 
         (q,) = qubit_manager.qalloc(self.n)
-        return cirq.H(q), {'q': np.array([q])}
+
+        if self.bit:
+            op = cirq.CircuitOperation(cirq.FrozenCircuit(cirq.X(q), cirq.H(q)))
```

Comment:
@mpharrigan `cirq.CircuitOperation(cirq.FrozenCircuit())` is essentially just an unnecessary boiler place here since it can be called in whatever utility method that calls `as_cirq_op`. This goes back to my original proposal of letting `as_cirq_op` be `as_cirq_optree` and wrap the yielded op-tree in a circuit operation at the call site when decomposing into a composite bloq. 

Can you clarify the benefits of this verbosity instead of the simplicity of my approach?

Reply:
I want to make sure we make it easy for users to express the fact that many times (like this one) you'd want to write a cirq conversion which is not 1:1. 

I tried to argue in my original comment https://github.com/quantumlib/Qualtran/pull/310#issuecomment-1634955959) that users can always wrap this conversion in a circuit operation, so there's really no way for us to enforce that we have a 1:1 mapping with a cirq (gate-)operation. It's good to see that now you also agree with me that the use case is a valid one and essentially end up doing the same thing.

Therefore, instead of pushing the burden of wrapping the OP-TREE in a circuit operation, we should start supporting an `as_cirq_optree` conversion.  The technical details of why we need to wrap the op-tree in a circuit operation should be handled by the library transparently from the user.

Reply:
I don't want decompositions hiding in the cirq-conversion code

Reply:
I dont think you can stop that from happening by requiring an extra boilerplate. The boilerplate does add friction to legit use cases like this one, where the decomposition is conceptually 1:1, but not exactly.